src/main/java/org/veupathdb/service/eda/ds/plugin/standalonemap/StandaloneMapMarkersPlugin.java

```diff
@@ -112,94 +104,55 @@ void addRow(double lat, double lon, String overlayValue) {
-    }    
-
-    // ranges must not overlap
-    // TODO again need to add support for dates here
-    if (anyHaveBinStart) {
-      Map<Double, Double> binEdges = new TreeMap<Double, Double>();
-      binRanges.forEach( bin -> binEdges.put(Double.parseDouble(bin.getBinStart()), Double.parseDouble(bin.getBinEnd())));
-      boolean first = true;
-      for (Double binStart : binEdges.keySet()) {
-        Double prevBinEnd;
-        if (first) {
-          prevBinEnd = binEdges.get(binStart);
-          first = false;
-        } else {
-          if (prevBinEnd > binStart) {
-            throw new ValidationException("Bin Ranges must not overlap");
-          }
-          prevBinEnd = binEdges.get(binStart);
-        }
-      }
+    int numDistinctOverlayVals = new HashSet<>(overlayConfig.getOverlayValues()).size();
+    if (numDistinctOverlayVals != overlayConfig.getOverlayValues().size()) {
+      throw new ValidationException("All overlay values must be unique: " + overlayConfig.getOverlayValues());
     }
   }
 
-  private static String recodeOverlayValue(String oldOverlayValue, List<BinRange> desiredOverlayValues) {
-    String newOverlayValue = oldOverlayValue;
-    // TODO assuming categorical vars will have labels but not starts and ends defined. maybe thats not explicit enough?
-    boolean isContinuous = desiredOverlayValues.stream().filter(bin -> bin.getBinStart() != null).findFirst().isPresent();
-
-    if (isContinuous) {
-      // TODO figure something else out for dates- not sure yet how to compare dates in java
-      newOverlayValue = desiredOverlayValues.stream().filter(bin -> (Double.parseDouble(bin.getBinStart()) < Double.parseDouble(oldOverlayValue) & Double.parseDouble(bin.getBinEnd()) > Double.parseDouble(oldOverlayValue))).findFirst().orElseThrow().getBinLabel();
-    } else {
-      boolean oldValueIsDesired = desiredOverlayValues.stream().filter(bin -> bin.getBinLabel().equals(oldOverlayValue)).findFirst().isPresent();
-      if (!oldValueIsDesired) {
-        newOverlayValue = "__UNSELECTED__";
-      }
+  private void validateContinousBinRanges(ContinousOverlayConfig overlayConfig) throws ValidationException {
+    if (!getUtil().getVariableDataShape(overlayConfig.getOverlayVariable()).equalsIgnoreCase(APIVariableDataShape.CONTINUOUS.toString())) {
+      throw new ValidationException("Input overlay variable %s is %s, but provided overlay configuration is for a conginuous variable"
+          .formatted(overlayConfig.getOverlayVariable().getVariableId(), getUtil().getVariableDataShape(overlayConfig.getOverlayVariable())));
+    }
+    boolean anyMissingBinStart = overlayConfig.getOverlayValues().stream().anyMatch(bin -> bin.getStart() == null);
+    boolean anyMissingBinEnd = overlayConfig.getOverlayValues().stream().anyMatch(bin -> bin.getEnd() == null);
+    if (anyMissingBinStart || anyMissingBinEnd) {
+      throw new ValidationException("All numeric bin ranges must have start and end.");
```

Comment:
overlapping ranges?

Reply:
Any bins I make will be.. and it doesn't seem crazy to ask custom bins to be. I thought I had added something for overlapping ranges already tbh